skimage/feature/_basic_features.py

```diff
@@ -83,16 +91,26 @@ def _mutiscale_basic_features_singlechannel(
         base=2,
         endpoint=True,
     )
-    with ThreadPoolExecutor(max_workers=num_workers) as ex:
-        out_sigmas = list(
-            ex.map(
-                lambda s: _singlescale_basic_features_singlechannel(
-                    img, s, intensity=intensity, edges=edges, texture=texture
-                ),
-                sigmas,
+    if threading_available:
+        with ThreadPoolExecutor(max_workers=num_workers) as ex:
```

Comment:
I think that would be nice to add (though I worry about the added complexity). I'll add this class and associated functions in the same file.

Reply:
Okay, after a few hours of reading the docs and some tries locally, I think I have a solution with [`contextlib.ExitStack`](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack) for now, so something like this should work:

```python
from contextlib import ExitStack, AbstractContextManager

# this is the same one as you defined above ⬆️
class _FallbackThreadPoolExecutor(AbstractContextManager):
    def __init__(self, max_workers):
        self.max_workers = max_workers

    def __enter__(self):
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        pass

    def map(self, fn, iterables):
        return map(fn, iterables)

# and then we can perform a conditional entry into the context manager here
threading_available = True
if (sys.platform == "emscripten") or (platform.machine() in ["wasm32", "wasm64"]):
    threading_available = False

if threading_available:
    executor_class = ThreadPoolExecutor
else:
    executor_class = _FallbackThreadPoolExecutor

# the exitstack can manage the closing of the context manager that gets created
# with stack.enter_context() here. not sure if it can be reduced further
with ExitStack() as stack:
    if threading_available:
        ex = stack.enter_context(ThreadPoolExecutor(max_workers=num_workers))
    else:
        ex = stack.enter_context(_FallbackThreadPoolExecutor(max_workers=num_workers))
    out_sigmas = list(
        ex.map(
            lambda s: _singlescale_basic_features_singlechannel(
                img, s, intensity=intensity, edges=edges, texture=texture
            ),
            sigmas,
        )
    )
features = itertools.chain.from_iterable(out_sigmas)
return features
```

This isn't the best way IMO, but if there is a better idea or if you have suggestions on this approach, I am happy to incorporate them. I haven't committed this change yet because I feel it is a (little) less readable in comparison to guarding the `threading` import and then proceeding with a non-threaded implementation without managing contexts.

Reply:
This seems a lot more complicated than what I had in mind. Perhaps you misunderstood? Let me investigate this locally and see if I can push something simpler..

Reply:
See https://github.com/scikit-image/scikit-image/pull/7350/commits/cdd9830199da49824ff5dd843ca18c1b0da3c0b9.

Reply:
Yes, I think so, too. An investigation from your end would be helpful :)